components/script/dom/webglrenderingcontext.rs

```diff
@@ -1281,15 +1290,26 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
             return self.webgl_error(InvalidOperation);
         }
 
-        if count <= 0 {
-            return self.webgl_error(InvalidOperation);
+        if count < 0 {
+            return self.webgl_error(InvalidValue);
         }
 
         if offset < 0 {
             return self.webgl_error(InvalidValue);
         }
 
-        if self.current_program.get().is_none() || self.bound_buffer_element_array.get().is_none() {
+        if self.current_program.get().is_none() {
```

Comment:
I have found this in WebGL spec: 

> If the CURRENT_PROGRAM is null, an INVALID_OPERATION error will be generated. WebGL performs additional error checking beyond that specified in OpenGL ES 2.0 during calls to drawArrays and drawElements.

Also required to pass some WebGL conformance tests.